pkg/kube/client/ssa/openapi_groupversion.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file is forked from upstream kubernetes/client-go
+https://github.com/kubernetes/client-go/blob/0b9a7d2f21befcfd98bf2e62ae68ea49d682500d/openapi/groupversion.go
+*/
+
+package ssa
+
+import (
+	"context"
+	"net/url"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kube-openapi/pkg/handler3"
+)
+
+// OpenAPIGroupVersion is the context-aware variant of openapi.GroupVersion
+// It also stores the ETag for the underlying discovery GV path
+type OpenAPIGroupVersion interface {
+	// Schema is the context-accepting variant of the upstream client-go implementation
+	Schema(ctx context.Context, contentType string) ([]byte, error)
+	ETag() string
```

Comment:
I don't understand why a schema loader and an arbitrary static etag are combined in one interface / struct.

Reply:
Ideally, we wanted to extract the ETag from the schema loader's HTTP response headers and use it for caching. 
So, something like
```
Schema(ctx context.Context, contentType string) (data []byte, etag string, err error)
```
However,  client-go `rest.Result` does not expose the response headers. 

Therefore, we "optimistically" associate the ETag that we got from the discovery with the schema loader, since we query the schema right after. In the case of, the schema content (the real ETag) changes in the short time between setting etag and loading schema, from functional correctness perspective we are not affected by storing with a "outdated etag" in the cache. We'll be using the latest correct schema,  but we'll just cause cache miss in the next reconciliation, since we always do discovery first and invalidate stales and re-query.
`ETag()` was interfaced per suggestion from previous review cycle. It makes sense to me to have it on another map, so that we associate them in our logic, not in the struct/interface.

Other alternative considered was to always query for Schema with `If-None-Match` header set to the extracted hash and rely on the API server returning 304, that hints us to use cached instance. This approach still requires extracting hash from discovery ofc.

Reply:
done https://github.com/crossplane-contrib/provider-kubernetes/pull/246/commits/9f522106676864f4c7527dffe1cf8533d8eb1434, Etag is removed from the interface and we return them as a separate map